components/Template.js

```diff
@@ -76,43 +81,32 @@ function transformData(fn, templateKey, originalData) {
 }
 
 function renderTemplate({template, compileOptions, helpers, data}) {
-  var hogan = require('hogan.js');
-  var forOwn = require('lodash/object/forOwn');
-  var content;
+  let isTemplateString = typeof template === 'string';
+  let isTemplateFunction = typeof template === 'function';
 
-  if (typeof template !== 'string' && typeof template !== 'function') {
+  if (!isTemplateString && !isTemplateFunction) {
     throw new Error('Template must be `string` or `function`');
+  } else if (isTemplateFunction) {
+    return template(data);
+  } else {
+    let transformedHelpers = transformHelpersToHogan(helpers, compileOptions);
+    let preparedData = {...data, helpers: transformedHelpers};
+    return hogan.compile(template, compileOptions).render(preparedData);
   }
+}
 
-  if (typeof template === 'function') {
-    content = template(data);
-  }
-
-  if (typeof template === 'string') {
-    data = addTemplateHelpersToData(data);
-
-    content = hogan.compile(template, compileOptions).render(data);
-  }
-
-  // We add all our template helper methods to the template as lambdas. Note
-  // that lambdas in Mustache are supposed to accept a second argument of
-  // `render` to get the rendered value, not the literal `{{value}}`. But
-  // this is currently broken (see
-  // https://github.com/twitter/hogan.js/issues/222).
-  function addTemplateHelpersToData(templateData) {
-    templateData.helpers = {};
-    forOwn(helpers, (method, name) => {
-      templateData.helpers[name] = function() {
-        return (text) => {
-          var render = (value) => hogan.compile(value, compileOptions).render(this);
-          return method.call(this, text, render);
-        };
-      };
+// We add all our template helper methods to the template as lambdas. Note
+// that lambdas in Mustache are supposed to accept a second argument of
+// `render` to get the rendered value, not the literal `{{value}}`. But
+// this is currently broken (see
+// https://github.com/twitter/hogan.js/issues/222).
+function transformHelpersToHogan(helpers, compileOptions) {
+  return mapValues(helpers, (method) => {
```

Comment:
Ok maybe we need to be careful about those refactoring we want to do. Because its then bring "noise" the PR. And it may be better code here but I fail to see it as easier to understand. Previous code was "good enough" I feel.